repo/hypre/package.py

```diff
@@ -18,6 +18,8 @@ class Hypre(BuiltinHypre):
         "gcc": "gnu",
     }
 
+    version("3.1_comm_cali", branch="comm_cali", submodules=False, git="https://github.com/gracenansamba/hypre.git")
```

Comment:
@gracenansamba It would be good to move the amg2023 changes to its own PR 
https://github.com/LLNL/benchpark/pull/318